packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -745,10 +745,19 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
     tsParseFunctionOrConstructorType(
       type: "TSFunctionType" | "TSConstructorType",
+      abstract?: boolean,
     ): N.TsFunctionOrConstructorType {
       const node: N.TsFunctionOrConstructorType = this.startNode();
       if (type === "TSConstructorType") {
-        this.expect(tt._new);
+        // $FlowIgnore
+        node.abstract = !!abstract;
+        if (abstract) {
+          // eat "abstract"
+          this.expectContextual("abstract");
+          this.expect(tt._new);
+        } else {
+          this.expect(tt._new);
+        }
```

Comment:
```suggestion
        if (abstract) this.expectContextual("abstract");
        this.expect(tt._new);
```

Or we could also just use `this.next()` instead of `this.expectContextual` since when `abstract` is `true` we know that there is the `abstract` keyword.